cmd/crates/soroban-spec-tools/src/contract.rs

```diff
@@ -113,6 +120,115 @@ impl Spec {
+                Version {
+                    encoding: Encoding::Module,
+                    ..
+                } => {}
+                Version {
+                    encoding: Encoding::Component,
+                    ..
+                } => {
+                    println!("components are not supported yet with the `strip` command");
+                    continue;
+                }
+
+                TypeSection(s) => section(SectionId::Type, s.range()),
+                ImportSection(s) => section(SectionId::Import, s.range()),
+                FunctionSection(s) => section(SectionId::Function, s.range()),
+                TableSection(s) => section(SectionId::Table, s.range()),
+                MemorySection(s) => section(SectionId::Memory, s.range()),
+                TagSection(s) => section(SectionId::Tag, s.range()),
+                GlobalSection(s) => section(SectionId::Global, s.range()),
+                ExportSection(s) => section(SectionId::Export, s.range()),
+                ElementSection(s) => section(SectionId::Element, s.range()),
+                DataSection(s) => section(SectionId::Data, s.range()),
+                StartSection { range, .. } => section(SectionId::Start, range),
+                DataCountSection { range, .. } => section(SectionId::DataCount, range),
+                CodeSectionStart { range, .. } => section(SectionId::Code, range),
+                CodeSectionEntry(_) => {}
+                wasmparser::Payload::CustomSection(c) => {
+                    println!("custom section: {:?}", c);
+
+                    module.section(&RawSection {
+                        id: SectionId::Custom as u8,
+                        data: &original_wasm_bytes[c.range()],
+                    });
+                }
+
+                ModuleSection { .. }
+                | InstanceSection(_)
+                | CoreTypeSection(_)
+                | ComponentSection { .. }
+                | ComponentInstanceSection(_)
+                | ComponentAliasSection(_)
+                | ComponentTypeSection(_)
+                | ComponentCanonicalSection(_)
+                | ComponentStartSection(_)
+                | ComponentImportSection(_)
+                | ComponentExportSection(_) => unimplemented!("component model"),
+
+                UnknownSection {
+                    id,
+                    contents,
+                    range: _,
+                } => {
+                    module.section(&RawSection { id, data: contents });
+                }
+
+                End(_) => {}
+                AliasSection(alias_section_reader) => todo!(),
+            }
+        }
+
+        // module.section(&RawSection {
+        //     id: SectionId::Custom as u8,
+        //     data: new_data,
+        // });
+
+        // then add the new custom section
+        module.section(&CustomSection {
+            name: Cow::Borrowed("contractmetav0"),
+            data: Cow::Borrowed(new_data),
+        });
```

Comment:
If the plan is to leave the existing custom section intact, and add a second custom section with the same name, can I suggest we use the wasmgen crate, which has the ability to simply add a new section without needing to parse or know anything about the existing sections?

An example of using it is here:
https://github.com/leighmcculloch/wasm-cs/blob/488eb67ef04f750ffb4ef1f2030d57cb5964af45/src/write.rs#L35

There might be a way to do the same with wasm-encoder. I'm just thinking it looks like we don't actually need to rewrite the existing module which seems like a risk if we don't support a section or there's a bug in that rewriting logic.

Reply:
Thank you for the example of wasmgen! 

I agree that rewriting the existing module does seem risky! Doing it that way in the PR was really helpful in understanding the tooling better, but I'll work on just appending a new section today.

Reply:
Just pushed up a change to use wasmgen, so much nicer! 🙏 

I'll plan to work on allowing a user to pass in meta as args to build, and testing next.